### PR TITLE
Add API doc for LDAP Sync

### PR DESCRIPTION
**Pull request by harry-kim**
Added documentation for LDAP Sync API Endpoint

Refs: #SDE-3432

### CODE REVIEW

File: source/includes/connections.md [hunk 1]
```diff
@@ -1,5 +1,114 @@
 # Connections
 
+## Get LDAP Connection for organization
+
+```http
+GET /api/v2/connections/ldap/ HTTP/1.1
+Accept: application/json
+Authorization: Token "YOUR SDE ACCESS TOKEN"
+
+{
```
Comment (RouganStriker) [inline source/includes/connections.md]: We should populate all the fields to make this example response as detailed as possible

```
"group_mapping": {
  "ldap_group1": "sde_group1",
  "ldap_group2": "sde_group2",
}
"ldap_filter": {"groups": ["LDAPGroup", "LDAPGroup3"], "users": ["user@example.org", "user2@example.org"]}
"user_schema": {
    "first_name": "gn",
    "last_name": "sn",
    "full_name": "cn",
    "email": "mail"
}
"deactivation": false,
"group_query": "(objectClass=group)"
"bind_dn": "cn=Administrator,cn=Users,dc=example,dc=org"
"base_dn": "dc=example,dc=org"
```

File: source/includes/connections.md [hunk 2]
```diff
@@ -1,5 +1,114 @@
...
+ldap_filter         | No       | Fine-grained control for users and groups during synchronization
+ldap_validate_cert  | No       | Determines whether or not to validate the SSL certificate for the LDAP server.
+user_schema         | No       | Define a custom user schema.
+deactivation        | No       | Auto-deactivate users
+bind_dn             | No       | The bind dn
+bind_password       | Yes      | The bind password
+page_size           | No       | Number of users to return per page.
+group_member_query  | No       | Gives the users of the specified group 
+group_query         | Yes      | Specify groups to return
+base_dn             | Yes      | The base dn
```
Comment (RouganStriker) [inline source/includes/connections.md]: Not required, will be computed from the bind_dn if not specified

File: source/includes/connections.md [hunk 3]
```diff
@@ -1,5 +1,114 @@
...
+ldap_server         | Yes      | The address of the LDAP server.
+ldap_method         | Yes      | The method used to connect to the LDAP server (ldap/ldaps).
+group_mapping       | No       | Object that maps LDAP to SDE groups.
+ldap_filter         | No       | Fine-grained control for users and groups during synchronization
+ldap_validate_cert  | No       | Determines whether or not to validate the SSL certificate for the LDAP server.
+user_schema         | No       | Define a custom user schema.
+deactivation        | No       | Auto-deactivate users
+bind_dn             | No       | The bind dn
+bind_password       | Yes      | The bind password
+page_size           | No       | Number of users to return per page.
```
Comment (RouganStriker) [inline source/includes/connections.md]: Should include the default value which is 1000

File: source/includes/connections.md [hunk 4]
```diff
@@ -1,5 +1,114 @@
...
+	"group_query": "<OU TO GET ALL GROUPS>",
+	"base_dn": "<BASE DN>"
+}
+```
+
+Fields              | Required | Description
+--------------------|----------|-------------
+ldap_server         | Yes      | The address of the LDAP server.
+ldap_method         | Yes      | The method used to connect to the LDAP server (ldap/ldaps).
+group_mapping       | No       | Object that maps LDAP to SDE groups.
```
Comment (RouganStriker) [inline source/includes/connections.md]: This is required

File: source/includes/connections.md [hunk 5]
```diff
@@ -1,5 +1,114 @@
...
+	"group_member_query": "(&(objectClass=user)(memberOf=%s))",
+	"group_query": "<OU TO GET ALL GROUPS>",
+	"base_dn": "<BASE DN>"
+}
+```
+
+Fields              | Required | Description
+--------------------|----------|-------------
+ldap_server         | Yes      | The address of the LDAP server.
+ldap_method         | Yes      | The method used to connect to the LDAP server (ldap/ldaps).
```
Comment (RouganStriker) [inline source/includes/connections.md]: Defaults to ldaps i think

Comment (harry-kim) [inline source/includes/connections.md]: updated

File: source/includes/connections.md (hunk 4 above)
Comment (harry-kim) [inline source/includes/connections.md]: updated

File: source/includes/connections.md (hunk 3 above)
Comment (harry-kim) [inline source/includes/connections.md]: updated

File: source/includes/connections.md (hunk 2 above)
Comment (harry-kim) [inline source/includes/connections.md]: updated

File: source/includes/connections.md [hunk 6]
```diff
@@ -1,5 +1,120 @@
...
+Fields              | Required | Description
+--------------------|----------|-------------
+ldap_server         | Yes      | The address of the LDAP server.
+ldap_method         | No       | The method used to connect to the LDAP server (defaults to ldaps).
+group_mapping       | Yes      | Object that maps LDAP to SDE groups.
+ldap_filter         | No       | Fine-grained control for users and groups during synchronization
+ldap_validate_cert  | No       | Determines whether or not to validate the SSL certificate for the LDAP server.
+user_schema         | No       | Define a custom user schema.
+deactivation        | No       | Auto-deactivate users
+bind_dn             | No       | The bind dn
```
Comment (JeffBain) [inline source/includes/connections.md]: I'd assume this is required, no? Given this is where we specify the user that the bind_password refers to etc.

**Comment by harry-kim** [comment]
can you resolve changes on github?

File: source/includes/connections.md [hunk 7]
```diff
@@ -1,5 +1,120 @@
...
+	"group_query": "(objectClass=group)",
+	"bind_dn": "cn=Administrator,cn=Users,dc=example,dc=org",
+	"base_dn": "dc=example,dc=org"
+}
+```
+Returns the LDAP connection associated to the organizaiton. 
+
+**`GET /api/v2/connections/ldap/`**
+
+---
```
Comment (RouganStriker) [inline source/includes/connections.md]: Can remove this line

File: source/includes/connections.md [hunk 8]
```diff
@@ -1,5 +1,120 @@
...
+}
+```
+
+Fields              | Required | Description
+--------------------|----------|-------------
+ldap_server         | Yes      | The address of the LDAP server.
+ldap_method         | No       | The method used to connect to the LDAP server (defaults to ldaps).
+group_mapping       | Yes      | Object that maps LDAP to SDE groups.
+ldap_filter         | No       | Fine-grained control for users and groups during synchronization
+ldap_validate_cert  | No       | Determines whether or not to validate the SSL certificate for the LDAP server.
```
Comment (RouganStriker) [inline source/includes/connections.md]: Defaults to True

File: source/includes/connections.md [hunk 9]
```diff
@@ -1,5 +1,120 @@
...
+user_schema         | No       | Define a custom user schema.
+deactivation        | No       | Auto-deactivate users
```
Comment (RouganStriker) [inline source/includes/connections.md]: Automatically deactivate groupless users in SDE

File: source/includes/connections.md (hunk 8 above)
Comment (harry-kim) [inline source/includes/connections.md]: updated

File: source/includes/connections.md (hunk 9 above)
Comment (harry-kim) [inline source/includes/connections.md]: updated

File: source/includes/connections.md [hunk 10]
```diff
@@ -1,5 +1,118 @@
...
+
+## Create an LDAP Connection
+
+```http
+POST /api/v2/connections/ldap/ HTTP/1.1
+Accept: application/json
+Authorization: Token "YOUR SDE ACCESS TOKEN"
+
+{
+	"ldap_server": "ldapServer:12345",
```
Comment (RouganStriker) [inline source/includes/connections.md]: Actually, all these params should be under "params".

The payload should be similar the AlmConnection and AnalysisConnection minus a few fields.

```
{
   "alias": "Example Analysis Connection",
    "connector": 2,
    "frequency": "daily",
    "params": {
        ...
    }
}
```

Comment (harry-kim) [inline source/includes/connections.md]: updated

**Comment by JeffBain** [comment]
+1. This stuff is looking good. Echoing Kelvin on holding off on version until we know which one this is going into, as it's unclear when we're cutting off v4.4.

**Pull request merged**